Documentation/Introduction.html

```diff
@@ -74,24 +74,47 @@
+Such applications are very difficult to build today.
+The development process demands expert level programmers and typically requires expensive iterations of the design and the architecture, as the workload grows.</p>
+<p>Most of today’s high scale properties are built as a composition of stateless n-tier services with most of application logic residing in the middle tier.</p>
+<p><img src="n-tier.png" alt=""></p>
+<p>While the model allows to scale out by adding more servers to the middle tier, it is constrained by the performance and scalability of the storage layer because most requests coming to the middle tier from the frontend web servers require one or more reads from storage, and updates are even more complicated and prone to concurrency issues and conflicts due to lack of coordination between the middle tier servers.
+It often requires caching in the stateless layer to get acceptable performance, adding complexity and introducing cache consistency issues.
+The other problem with the stateless n-tier model is that it doesn&#39;t support well horizontal communications between individual application entities exposed by the middle tier, which makes it hard to implement complex business logic with multiple entities performing individual operations as part of processing a request.</p>
+<h2 id="orleans-as-a-stateful-middle-tier">Orleans as a Stateful Middle Tier</h2>
+<p>Orleans provides an intuitive way of building a stateful middle tier, where various business logic entities appear as sea of isolated globally addressable .NET objects (grains) of different application defined types distributed across a cluster of servers (silos).</p>
+<p><img src="actor-middle-tier.png" alt=""></p>
+<p>A grain type is a simple .NET class that implements one or more application defined grain interfaces.
+Individual grains are instances of application defined grain classes that get automatically created by the Orleans runtime on servers on an as-needed basis to handle requests for those grains.
+Grains naturally map to most application entities, such as users, devices, sessions, inventories, orders, etc., which makes it very easy to build business logic that is object-oriented but scales transparently across a cluster of servers.
+Each grain has a stable logical identity (key) within its grain type chosen by the application logic, for example, user email or device ID or inventory SKU code.
+Orleans guarantees single-threaded execution of each individual grain, hence protecting the application logic from perils of concurrency and races.</p>
+<h2 id="grain-lifecycle">Grain Lifecycle</h2>
+<p>Grain can have persistent state in storage or in-memory state or a combination of both.
+Any grain can be called by any other grain or by a frontend (client) without the need to create or instantiate the target grain.
+The Orleans programming model makes grains appear as if they are in memory the whole time.
```

Comment:
I think it is worth to call out that grain call other grain by referencing its logical identity. Otherwise it feels like missing a how here.
For example, the above line can be: 
_Without the need to create or instantiate the target grain, any grain can be called by any other grain or by a frontend (client) with referencing the grain's logical identity._